### DISCUSSION TITLE
2 SML Meters at one Wemos are not working

### DISCUSSION THREAD

**Discussion by derrodi** [arendst/Tasmota#16624] [Q&A]
Hey everybody,
just need some help. I have 1 Wemos D1 Mini and two IR Readings (https://www.ebay.de/itm/353940190756) Booth are directly connected to the Wemos. I use this script

`>D
>B
=>sensor53 r
>M 2
+1,3,o,0,300,Strom,1,60,2F3F210D0A,063030300D0A
+2,14,o,0,300,Stromwp,12,30,2F3F210D0A,063030300D0A
1,1.8.0*00(@1,Gesamtverbrauch,KWh,Pges,2
2,1.8.1(@1,HT,KWh,Pges3,2
2,1.8.2(@1,NT,KWh,Pges1,2

#
`
I can read out each meter for its own. But if i put both meter in one Script i only get the values for the first meter.

**Comment by Andre1Becker** [comment]
Hi @derrodi,
it seems you had a typo in your script. Here is an example of my current script which you could easly adapt:

```
;defines numbers of Meters
>M 2 
;defines type of Meter ONE - in this case a Power meter - note the +1  and the first 1 on the next lines!
+1,3,s,0,9600,SML 
1,77070100010800ff@1000,Verbrauch,KWh,Total_in,1
1,77070100100700ff@1,Aktueller Verbrauch,W,Power_curr,1

;defines type of Meter TWO- in this case a GAS meter - note the +2 and the first 2 on the next line!
+2,5,c,1,-25,GAS
2,1-0:1.8.0*255(@100,Zaehlerstand,m³,GAS_Total,18
#
```

the MQTT output should be similar to this:

`tele/sensorsm01/SENSOR = {"Time":"2022-12-04T13:06:16","SML":{"Total_in":43970.4,"Power_curr":624.0},"GAS":{"GAS_Total":23453.00}}`


Also if you want to define the start value of the counter, you could do that by open the console and enter:

`sensor53 c1 752389`
This should result in something like this:

```
MQT: tele/sensorsm01/SENSOR = {"Time":"2022-12-04T13:36:14","GAS":{"GAS_Total":7523.89}}
MQT: stat/sensorsm01/RESULT = {"Time":"2022-12-04T13:36:14","SML":{"CMD":"counter1: 752389"}}
```

Hope this helps,

Cheers!
